lyngk.js

```diff
@@ -0,0 +1,179 @@
+        colorCount.Blue = colorCount.Blue - 1;
+      }
+      else if (k < 24) {
+        drawGamePiece((xRoot * coordinates[0] + padding), (yOffset * coordinates[1] + padding), "Red");
+        colorCount.Red = colorCount.Red - 1;
+      }
+      else if (k < 32) {
+        drawGamePiece((xRoot * coordinates[0] + padding), (yOffset * coordinates[1] + padding), "Green");
+        colorCount.Green = colorCount.Green - 1;
+      }
+      else if (k < 40) {
+        drawGamePiece((xRoot * coordinates[0] + padding), (yOffset * coordinates[1] + padding), "Black");
+        colorCount.Black = colorCount.Black - 1;
+      }
+      else {
+        drawGamePiece((xRoot * coordinates[0] + padding), (yOffset * coordinates[1] + padding), "White");
+        colorCount.White = colorCount.White - 1;
+      }
+    }
+    stackGamePiece((xRoot * boardPosition['E4'][0] + padding), (yOffset * boardPosition['E4'][1] + padding), 'Black', 2);
+    stackGamePiece((xRoot * boardPosition['E4'][0] + padding), (yOffset * boardPosition['E4'][1] + padding), 'Blue', 3);
+    stackGamePiece((xRoot * boardPosition['E4'][0] + padding), (yOffset * boardPosition['E4'][1] + padding), 'Green', 4);
+    stackGamePiece((xRoot * boardPosition['E4'][0] + padding), (yOffset * boardPosition['E4'][1] + padding), 'White', 5);
+    stackGamePiece((xRoot * boardPosition['E4'][0] + padding), (yOffset * boardPosition['E4'][1] + padding), 'Ivory', 6);*/
+  }
+
+  function drawLine(x1, y1, x2, y2) {
+	  ctx.beginPath();
+	  ctx.moveTo(x1, y1);
+	  ctx.lineTo(x2, y2);
+	  ctx.stroke();
+  }
+
+  function drawGamePiece(xcenter, ycenter, color) {
+    ctx.beginPath();
+    ctx.arc(xcenter, ycenter, 10, 0, 2*Math.PI);
+    ctx.fillStyle = colorValues[color];
+    ctx.fill();
+    ctx.stroke();
+  }
+
+  function stackGamePiece(xcenter, ycenter, color, stackCount) {
+    stackOffset = (stackCount - 1) * 5;
+    drawGamePiece(xcenter, ycenter - stackOffset, color);
+  }
+
+  function shuffleArray(input) {
+    var inputArray = input;
+    for(var i = inputArray.length; i > 0; i--) {
+      randomIndex = Math.random() * (i - 1);
+      x = inputArray[randomIndex];
+      inputArray[randomIndex] = inputArray[i];
+      inputArray[i] = x;
+    }
+    return inputArray;
+  }
+
+  function translateBoardString(boardString) {
+    var boardArray = boardString.split('|');
+    for (var i = 0; i < boardArray.length; i++) {
+      for (var j = 0; j < boardArray[i].length; j++) {
+        var boardIndex = nodeList[i];
+        var color = letterToColor[boardArray[i][j]];
+        if (color != "Empty") {
+          stackGamePiece((xRoot * boardPosition[boardIndex][0] + padding), (yOffset * boardPosition[boardIndex][1] + padding), color, j + 1);
+        }
+      }
+    }
+  }
+
+  translateBoardString('-|I|GJ|-|JIGK|I|B|GR|-|K|R|-|-|I-G|I|BJ|K|-R|-|-|G|-|-|B|-|-|RK|-|R|BIG|-R|-|B|-|I|I|-|K|-|G-|BK|K|RGKB|BR');

```

Comment:
Declare a variable kTestString or some such and put this up top with the other constants.